hooks/persistentvolumeclaim.go

```diff
@@ -0,0 +1,102 @@
+package hooks
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/go-logr/logr"
+	"github.com/topolvm/pvc-autoresizer/runners"
+	admissionv1 "k8s.io/api/admission/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+//+kubebuilder:webhook:path=/pvc/mutate,mutating=true,failurePolicy=fail,sideEffects=None,groups="",resources=persistentvolumeclaims,verbs=create,versions=v1,name=mpersistentvolumeclaim.topolvm.io,admissionReviewVersions={v1}
+
+type persistentVolumeClaimMutator struct {
+	apiReader client.Reader
+	dec       *admission.Decoder
+	log       logr.Logger
+}
+
+var _ admission.Handler = &persistentVolumeClaimMutator{}
+
+func (m *persistentVolumeClaimMutator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	if req.Operation != admissionv1.Create {
+		return admission.Allowed("not a Create request")
+	}
+	pvc := &corev1.PersistentVolumeClaim{}
+	if err := m.dec.Decode(req, pvc); err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+	groupLabelKey, ok := pvc.Annotations[runners.InitialResizeGroupByAnnotation]
+	if !ok || groupLabelKey == "" {
+		return admission.Allowed("annotation not set")
+	}
+	group, ok := pvc.Labels[groupLabelKey]
+	if !ok || group == "" {
+		return admission.Errored(http.StatusBadRequest, fmt.Errorf("no value is set to the label key %s", groupLabelKey))
+	}
+
+	storageLimit, err := runners.PvcStorageLimit(pvc)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+	if storageLimit.IsZero() {
+		return admission.Allowed("ignore the PVC because it has no storage limit annotation")
+	}
+
+	pvcList := &corev1.PersistentVolumeClaimList{}
+	err = m.apiReader.List(ctx, pvcList, &client.ListOptions{
+		Namespace:     pvc.Namespace,
+		LabelSelector: labels.SelectorFromSet(map[string]string{groupLabelKey: group}),
+	})
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+
+	requestedSize := pvc.Spec.Resources.Requests[corev1.ResourceStorage]
+	newSize := requestedSize
+	for _, item := range pvcList.Items {
+		if item.Spec.Resources.Requests.Storage().Cmp(newSize) > 0 {
+			newSize = item.Spec.Resources.Requests[corev1.ResourceStorage]
```

Comment:
It can be simplified as:
```suggestion
		if itemSize := item.Spec.Resources.Requests.Storage(); itemSize.Cmp(newSize) > 0 {
			newSize = itemSize
```

Reply:
Fixed.